src/textual/_xterm_parser.py

```diff
@@ -129,25 +131,42 @@ def parse(self, on_token: TokenCallback) -> Generator[Awaitable, str, None]:
                         if len(peek_buffer) == 1 and not more_data():
                             continue
 
+                # Look ahead through the suspected escape sequence for a match
                 while True:
-                    sequence += yield read1()
+
+                    # If we run into another ESC at this point, then we've failed
+                    # to find a match, and should issue everything we've seen within
+                    # the suspected sequence as Key events instead.
+                    buffer = yield self.peek_buffer()
+                    if (
+                        buffer
+                        and buffer[0] == ESC
+                        or len(sequence) > _MAX_SEQUENCE_SEARCH_THRESHOLD
+                    ):
+                        for character in sequence:
+                            keys = get_key_ansi_sequence(character, None)
```

Comment:
I suspect that if we have an `ESC` to be sent from an unrecognised sequence, we should send a `^`  since other wise it would be seen as the user hitting the literal escape key.

Reply:
Yeah, you might be right. I was thinking that we might not know if it was literally the user pressing an escape key followed by some data that doesn't match an ANSI sequence.

Reply:
Have made it such that when we perform the backtracking, `\x1b` gets converted to `^` rather than `escape`.